Slim/Web/Settings/Player/Basic.pm

```diff
@@ -204,7 +204,7 @@ sub getVisualModes {
 	my $modes  = $client->display->visualizerModes;
 	my $nmodes = $client->display->visualizerNModes;
 
-	for (my $i = 0; $i <= $nmodes; $i++) {
+	for (my $i = 0; $i < $nmodes; $i++) {
```

Comment:
I'm not sure I do understand what you're trying to fix here. With this change applied I'm missing  one of the visualizer settings. I'd have two empty items at the bottom.  What issue are you trying to address?

Reply:
Ah ... what seems to me is that the loop is going 0...#$modes, so one index beyond the size of the visu modes and that means that the last ```my $desc = $modes->[$i]{'desc'};``` does not point to anything meaninful. 

It caused me a problem in the squeezeesp32 plugin, where ->modes() is not anymore a subroutine returning \@modes, but I need every player to have a different set of modes, modes() is an accessor to an array that is bein built when the payer is created.

When using that accessor out of bounds, then getPlayingDisplayModes whines that I'm using an undefined value as an array in ```for (my $j = 0; $j < scalar @$desc; $j++)```

Reply:
Hmm... that check about the Transporter display on line 195. Should you even reach that code with a boom? Or your own player model? I did test with a Transporter.

Reply:
I'm sorry @mherger I don't see the connection with line 195 of Transporter's display. That patch is for Web/Settings/Player/Basic.pm and that function (getPlayingDisplayModes) is unique in the whole LMS system. The Transporter has 6 display modes, less than the Boom.

Reply:
Wouldn't line 195 cause any call to return quickly if it hadn't a Transporter type display?

Reply:
`getPlayingDisplayModes` is further up the code, isn't it?

Reply:
Oh I see - we are not talking about the same function. I’m taking about getPlayingDisplayMode, not about geVisualModes.
Although I would fail to understand why going over the array boundaries is correct and needed in getVisualModes

Reply:
I'm talking about your suggested change which is in `getVisualModes `, while you refer to `getPlayingDisplayMode`. Did you close the wrong PR? I'm utterly confused.

Reply:
`$#modes` is not the number of items, but the index of the last item. Therefore we don't go past the list's length.

Reply:
oh crap ... 

1- I did the change manually and not by a diff hence I patched the wrong function
2- I totally forgot (I almost never use it) that $# is last index, not size and for some reasons, in my code, it was replaced by scalar (x) so ...

Very sorry for the confusion :-( I'm going to hang myself somewhere